src/usr/local/captiveportal/index.php

```diff
@@ -82,7 +82,13 @@
 } elseif (!empty($cpsession) && (!isset($_POST['logout_id']) || !isset($config['captiveportal'][$cpzone]['logoutwin_enable']))) {
 	/* If client try to access captive portal page while already connected, 
 		but no custom logout page does exist and logout popup is disabled */	
-	echo gettext("You are connected.");
+	if ($_REQUEST['voucher'] && $_REQUEST['redirurl']) {
+		header("Location: {$redirurl}");
+	} elseif ($_REQUEST['voucher'] && !empty($cpcfg['redirurl'])) {
+		header("Location: {$cpcfg['redirurl']}");
```

Comment:
Allowing a redirect to a URL based on a GET request like this is a potential path for abuse/exploitation.

By making the user to a POST, we get CSRF protection and other similar assurances about the origin of the request.

If this only happens via POST, then change `$_REQUEST` to `$_POST`. Or make separate variations for `$_POST` and `$_GET` and have the `$_GET` version behave differently (such as landing on a page with the full redirect URL printed out and linked).

Reply:
Fixed
Now it shows "You can proceed to <URL>" link